lib/openstax/rescue_from/deploy_utils.rb

```diff
@@ -0,0 +1,18 @@
+module OpenStax
+  module RescueFrom
+    class DeployUtils
```

Comment:
Similar to how we want to move specific email addresses out of the gem, we will want to move this code out of the gem too.  More than OpenStax specific, it is Tutor specific so won't work when we get to `exchange-dev.openstax.org`, etc.  

The `server_nickname` can be configured with a `server_nickname_proc`.  The installed config could look like:

```ruby
# set with a proc that returns a server nickname to add to log messages
# if returns nil, "unknown_deploy" will be used
# config.server_nickname_proc = Proc.new { nil }
```

Tutor's configuration file could have the Tutor-specific details

```ruby
config.server_nickname_proc = Proc.new { 
  url = Rails.application.secrets.mail_site_url
  return nil if url.nil?
  match = url.match(/\Atutor-(.+)\.openstax/)

  if url == 'tutor.openstax.org'
    'production'
  elsif match && match[1]
    match[1]
  else
    url
  end
}